EIPS/eip-6733.md

```diff
@@ -0,0 +1,71 @@
+---
+eip: 6733
+title: Revert creation in case of collision
+description: Revert contract creation if address already has code
+author: Renan Rodrigues de Souza (@RenanSouza2)
+discussions-to: https://ethereum-magicians.org/t/eip-revert-on-address-collision/13442
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-03-20
+---
+
+## Abstract
+
+There is no current definition to what happens when a contract creation happens in an address with code already deployed. This fix prevents an attack of deploying contract code and later changing the code arbitrarily.
+
+## Motivation
+
+In [EIP-3607](./eip-3607.md) it was estimated that to create an address collision it would take something about 10 billion USD and this number decreases as computers processing power grow.
+
+## Specification
+
+In contract creation where `new contract address` has a `CODEHASH != EMPTYCODEHASH` MUST have its current environment execution reverted, where `EMPTYCODEHASH = 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470`.
```

Comment:
This is insufficient.

Also, shouldn't you add a 100 gas cost increase for the equivalent of `extcodesize`?

```suggestion
## Specification

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.

If a previously-existing account in the state trie would have its code modified by the the `CREATE` or `CREATE2` opcodes, or by transferring ether to the zero address, instead, the call or ether transfer MUST revert. The `CREATE` and `CREATE2` opcodes MUST cost an additional `100` gas.
```

Reply:
I am not understanding why include the transfer to the zero address. Do you mean a transfer to  ∅ address?

Reply:
Zero address = $\varnothing$ address